source/game_sa/Hud.cpp

```diff
@@ -505,14 +818,100 @@ void CHud::DrawOddJobMessage(uint8 priority) {
+    
+    CRadar::DrawMap();
+
     CRect rect;
-    rect.left   = SCREEN_STRETCH_X(54.0f);
-    rect.top    = SCREEN_HEIGHT - SCREEN_STRETCH_Y(104.0f) - SCREEN_STRETCH_Y(85.0f);
-    rect.right  = SCREEN_STRETCH_X(64.0f) + SCREEN_STRETCH_X(54.0f);
-    rect.bottom = SCREEN_STRETCH_Y(64.0f) + SCREEN_HEIGHT - SCREEN_STRETCH_Y(104.0f) - SCREEN_STRETCH_Y(85.0f);
-    Sprites[5].Draw(rect, CRGBA(255, 255, 255, 255));
+    CVehicle* playerVehicle= FindPlayerVehicle(-1, 0);
+    CPlayerPed* PlayerPed = FindPlayerPed(-1);
+
+    // todo Use eRadarMode::BLIPS_ONLY in new commits
+    if (FrontEndMenuManager.m_nRadarMode != 1) {
+        if (playerVehicle &&
+            (FindPlayerVehicle(-1, 0)->m_nVehicleSubType == VEHICLE_TYPE_PLANE &&
+            FindPlayerVehicle(-1, 0)->m_nModelIndex != MODEL_VORTEX)) {
+            if (PlayerPed->m_aWeapons[PlayerPed->m_nActiveWeaponSlot].m_nType != WEAPON_PARACHUTE) {
+                // Non reverse code
+                float angle = FindPlayerHeading(0) - (CRadar::m_fRadarOrientation + M_PI);
+                // printf("angle %f\n",angle);
+                // CMatrixLink m_matrix = playerVehicle->GetMatrix();
+                // vector_constructor_403D60((char*)&rect, 8, 4, (int(__thiscall*)(_DWORD))CVector2D::CVector2D);
+
+                // long double v2 = -atan2(-m_matrix.GetRight().z, m_matrix.GetUp().z) - 0.78539819;
+                /* for (int i = 0; i < 4; i++) {
+                     *(&v66 + 2 * i) = sin(i * 1.5707964 + v2) * SCREEN_STRETCH_X(76.0) + SCREEN_STRETCH_X(87.0);
+                    *(&v67 + 2 * i) = cos(i * 1.5707964 + v2) * SCREEN_SCALE_Y(76.0) + SCREEN_STRETCH_FROM_BOTTOM(84.0f);
+                 }*/
+                CRadar::DrawRotatingRadarSprite(
+                    &Sprites[4],
+                    SCREEN_STRETCH_X(86.0f),
+                    SCREEN_STRETCH_FROM_BOTTOM(67.0f),
+                    angle,
+                    SCREEN_STRETCH_X(76.0f),
+                    SCREEN_STRETCH_Y(63.0f),
+                    CRGBA(255, 255, 255, 255)
+                );
+            }
+            /*if (!playerVehicle || playerVehicle->m_nVehicleSubType != VEHICLE_TYPE_PLANE &&
+                playerVehicle->m_nVehicleSubType != VEHICLE_TYPE_HELI ||
+                playerVehicle->m_nModelIndex == MODEL_VORTEX) {
+            }*/
+
+            // Padding from my head
+            rect.left = SCREEN_STRETCH_X(20.0);
+            rect.top = SCREEN_STRETCH_FROM_BOTTOM(104.0);
+            rect.right = SCREEN_STRETCH_X(30.0);
+            rect.bottom = SCREEN_STRETCH_FROM_BOTTOM(28.0);
+            CSprite2d::DrawRect(rect, CRGBA(10, 10, 10, 100));
+        }
+    }
+    // ring orig padding +/- 3.0f
+
+
+    // top left
+    rect.left = SCREEN_STRETCH_X(36.0);
+    rect.top = SCREEN_STRETCH_FROM_BOTTOM(108.0);
+    rect.right = SCREEN_STRETCH_X(87.0);
+    rect.bottom = SCREEN_STRETCH_FROM_BOTTOM(66.0);
+    Sprites[3].Draw(rect, CRGBA(0, 0, 0, 255));
```

Comment:
!!! check rect properties before applying

```suggestion
    rect = {
        SCREEN_STRETCH_X(36.0);
        SCREEN_STRETCH_FROM_BOTTOM(108.0);
        SCREEN_STRETCH_X(87.0);
        SCREEN_STRETCH_FROM_BOTTOM(66.0);
    }
    Sprites[3].Draw(rect, CRGBA(0, 0, 0, 255));
```

Reply:
ditto on top of this